### PR TITLE
Utility for file encryption and decryption on unix machines

### PR DESCRIPTION
**Pull request by Jauler**
I have written a little utility for files encryption and decryption on unix machine. Please have a look :).

### CODE REVIEW

File: file_crypt.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+	//Reset parameters
+	memset(&params, 0, sizeof(params));
+
+	//Parse arguments
+	if((status = parse_args(argc, argv, &params)))
+		goto CLEANUP;
+
+	//try to open input file
+	in_f = fopen(params.input_filename, "r");
```
Comment (denisdemaisbr) [inline file_crypt.c]: change to "rb" (read-only binary)

File: file_crypt.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	if(in_f == NULL){
+		fprintf(stderr, "Could not open input file\n");
+		status = -EPERM;
+		goto CLEANUP;
+	}
+
+	//Try to open output file
+	out_f = fopen(params.output_filename, "w");
```
Comment (denisdemaisbr) [inline file_crypt.c]: change to "wb" (write binary)

**Comment by Jauler** [comment]
When I wrote those changes (few years ago) I changed initialisation vector storage place. But now I think, that this is a bad Idea, as there is no way to reset it for new encryption session, so I removed that commit.

**Comment by tricong** [comment]
How can i read hexa string from file01 then encrypt them using AES128_ECB_encrypt(). After that, write encypted result to file02. I am a newbie. Plz help me. Thank u.

**Comment by Jauler** [comment]
Would you like to convert that hex string to binary? Or encrypt string as is?

**Comment by tricong** [comment]
i mean encrypt hex string, example "6bc1bee22e409f96e93d7e117393172aae2d8a571e03ac9c9eb76fac45af8e5130c81c46a35ce411e5fbc1191a0a52eff69f2445df4f9b17ad2b417be66c3710", my file has alot line "6bc1....3710".

**Comment by tricong** [comment]
From tiny-AES-128 program I have to compute the execution time of encryption and decryption in ECB and CBC mode. Do you have any solution to help me?

**Pull request closed without merging**